### ISSUE TITLE
🐛 [performance] iOS crash app when HttpMetric stops

### ISSUE THREAD

**Issue by CarlosVallsCiscar** [FirebaseExtended/flutterfire#7523]
## Bug report

**Describe the bug**
I have updated performance to the latest version (from 0.7.1+4 to 0.8.0+2) and now when measuring an HTTPRequest the app crash on iOS (emulator and real device) and not on Android.

### Steps to reproduce

n/a

### Expected behavior

n/a

### Sample project

n/a

## Additional context

**Real device crash trace:**

```
-[NSNull length]: unrecognized selector sent to instance 0x1dbf8c558
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull length]: unrecognized selector sent to instance 0x1dbf8c558'
*** First throw call stack:
(0x18182504c 0x199e99f54 0x181902014 0x1817ba474 0x1817b95b0 0x181facbfc 0x181faa784 0x181fe7a94 0x18203699c 0x1046560a0 0x10314b450 0x10314b0bc 0x10314ac00 0x10763195c 0x107182130 0x1074feafc 0x10742c214 0x10742ff6c 0x18183a268 0x1817bec40 0x1817b943c 0x181797c58 0x1817ab3b8 0x19d13b38c 0x18414b6a8 0x183eca7f4 0x102a4cbdc 0x103fd5a24)
libc++abi: terminating with uncaught exception of type NSException
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001b866c964 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1b866c964 <+8>:  b.lo   0x1b866c984               ; <+40>
    0x1b866c968 <+12>: pacibsp 
    0x1b866c96c <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1b866c970 <+20>: mov    x29, sp
```

**Emulator crash trace:**
```
Application Specific Information:
dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot
CoreSimulator 776.4 - Device: iPhone 13 Pro Max (********-****-****-****-************) - Runtime: iOS 15.0 (19A339) - DeviceType: iPhone 13 Pro Max
terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull length]: unrecognized selector sent to instance 0x7fff8a349fb0'
dyld4 config: DYLD_ROOT_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot
abort() called


Last Exception Backtrace:
0   CoreFoundation                	    0x7fff203fbba4 __exceptionPreprocess + 226
1   libobjc.A.dylib               	    0x7fff2019ebe7 objc_exception_throw + 48
2   CoreFoundation                	    0x7fff2040a821 +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
3   CoreFoundation                	    0x7fff204000bc ___forwarding___ + 1433
4   CoreFoundation                	    0x7fff204021e8 _CF_forwarding_prep_0 + 120
5   CFNetwork                     	    0x7fff23f9a87e 0x7fff23dad000 + 2021502
6   CFNetwork                     	    0x7fff23e834e7 0x7fff23dad000 + 877799
7   CFNetwork                     	    0x7fff23e83840 0x7fff23dad000 + 878656
8   CFNetwork                     	    0x7fff23dc56de 0x7fff23dad000 + 100062
9   FirebasePerformance           	       0x105c8f38e -[FIRHTTPMetric stop] + 750 (FIRHTTPMetric.m:129)
10  Runner                        	       0x10476ec42 -[FLTHttpMetric stop:result:] + 802
11  Runner                        	       0x10476e86a -[FLTHttpMetric handleMethodCall:result:] + 282
12  Runner                        	       0x10476e364 -[FLTFirebasePerformancePlugin handleMethodCall:result:] + 436
13  Flutter                       	       0x108fbd8a1 __45-[FlutterMethodChannel setMethodCallHandler:]_block_invoke + 104
14  Flutter                       	       0x108a66c4d flutter::PlatformMessageRouter::HandlePlatformMessage(std::__1::unique_ptr<flutter::PlatformMessage, std::__1::default_delete<flutter::PlatformMessage> >) const + 193
15  Flutter                       	       0x108a6c307 flutter::PlatformViewIOS::HandlePlatformMessage(std::__1::unique_ptr<flutter::PlatformMessage, std::__1::default_delete<flutter::PlatformMessage> >) + 35
16  Flutter                       	       0x108e976c9 std::__1::__function::__func<fml::internal::CopyableLambda<flutter::Shell::OnEngineHandlePlatformMessage(std::__1::unique_ptr<flutter::PlatformMessage, std::__1::default_delete<flutter::PlatformMessage> >)::$_16>, std::__1::allocator<fml::internal::CopyableLambda<flutter::Shell::OnEngineHandlePlatformMessage(std::__1::unique_ptr<flutter::PlatformMessage, std::__1::default_delete<flutter::PlatformMessage> >)::$_16> >, void ()>::operator()() + 81
17  Flutter                       	       0x108d9c866 fml::MessageLoopImpl::FlushTasks(fml::FlushType) + 164
18  Flutter                       	       0x108da2d18 fml::MessageLoopDarwin::OnTimerFire(__CFRunLoopTimer*, fml::MessageLoopDarwin*) + 26
19  CoreFoundation                	    0x7fff2036a6c6 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
20  CoreFoundation                	    0x7fff2036a1c2 __CFRunLoopDoTimer + 923
21  CoreFoundation                	    0x7fff20369781 __CFRunLoopDoTimers + 265
22  CoreFoundation                	    0x7fff20363dc0 __CFRunLoopRun + 2010
23  CoreFoundation                	    0x7fff20363103 CFRunLoopRunSpecific + 567
24  GraphicsServices              	    0x7fff2c851cd3 GSEventRunModal + 139
25  UIKitCore                     	    0x7fff24ffbe63 -[UIApplication _run] + 928
26  UIKitCore                     	    0x7fff25000a53 UIApplicationMain + 101
27  Runner                        	       0x10405053f main + 63 (AppDelegate.swift:6)
28  dyld_sim                      	       0x105463e1e start_sim + 10
29  dyld                          	       0x105e234fe start + 462
30  ???                           	               0x0 ???
31  Runner                        	       0x10404a000 ???
```
---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-x64, locale es-ES)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[☠] IntelliJ IDEA Ultimate Edition (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know about this issue at https://github.com/flutter/flutter/issues.
    ✗ FormatException: Unexpected extension byte (at offset 5)
[✓] VS Code (version 1.62.3)
[✓] Connected device (3 available)
    ! Error: iPhone is busy: Making the device ready for development. Xcode will continue when iPhone is finished. (code -10)

! Doctor found issues in 1 category.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.15.0
Flutter SDK 2.8.0
agrarware 1.2.0+16

dependencies:
- badges 2.0.2 [flutter]
- cached_network_image 3.2.0 [flutter flutter_cache_manager octo_image cached_network_image_platform_interface cached_network_image_web]
- camera 0.9.4+5 [camera_platform_interface camera_web flutter pedantic quiver flutter_plugin_android_lifecycle]
- chopper 4.0.3 [http meta logging]
- cupertino_icons 1.0.4
- device_info_plus 3.1.1 [flutter device_info_plus_platform_interface device_info_plus_linux device_info_plus_macos device_info_plus_web device_info_plus_windows]
- file_picker 4.2.7 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path]
- firebase_analytics 9.0.2 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.10.5 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 2.4.3 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_messaging 11.2.3 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_performance 0.8.0+2 [firebase_core firebase_core_platform_interface firebase_performance_platform_interface firebase_performance_web flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_app_badger 1.3.0 [flutter]
- flutter_cache_manager 3.3.0 [clock collection file flutter http path path_provider pedantic rxdart sqflite uuid]
- flutter_image_compress 1.1.0 [flutter]
- flutter_local_notifications 9.1.5 [clock flutter flutter_local_notifications_linux flutter_local_notifications_platform_interface timezone]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_platform_widgets 1.11.1 [flutter]
- flutter_secure_storage 5.0.2 [meta flutter flutter_secure_storage_linux flutter_secure_storage_macos flutter_secure_storage_platform_interface flutter_secure_storage_web flutter_secure_storage_windows]
- flutter_speed_dial 4.6.6 [flutter]
- flutter_webview_plugin 0.4.0 [flutter]
- font_awesome_flutter 9.2.0 [flutter]
- fuzzy 0.4.0-nullsafety.0 [latinize]
- get_it 7.2.0 [async collection]
- google_maps_flutter 2.1.1 [flutter flutter_plugin_android_lifecycle google_maps_flutter_platform_interface]
- image_cropper 1.4.1 [flutter]
- image_picker 0.8.4+4 [flutter flutter_plugin_android_lifecycle image_picker_for_web image_picker_platform_interface]
- infinite_scroll_pagination 3.1.0 [flutter sliver_tools]
- intl 0.17.0 [clock path]
- json_annotation 4.4.0 [meta]
- location 4.3.0 [flutter location_platform_interface location_web]
- logging 1.0.2
- native_pdf_view 5.0.0 [flutter photo_view synchronized native_pdf_renderer]
- nfc_manager 3.1.0 [flutter]
- open_file 3.2.1 [flutter ffi]
- package_info_plus 1.3.0 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- path_provider 2.0.7 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- permission_handler 8.3.0 [flutter meta permission_handler_platform_interface]
- photo_view 0.13.0 [flutter]
- rxdart 0.27.3
- share_plus 3.0.4 [meta mime flutter share_plus_platform_interface share_plus_linux share_plus_macos share_plus_windows share_plus_web]
- shared_preferences 2.0.9 [flutter meta shared_preferences_android shared_preferences_ios shared_preferences_linux shared_preferences_macos shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- timelines 0.1.0 [flutter]
- url_launcher 6.0.17 [flutter meta url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- webview_flutter 3.0.0 [flutter webview_flutter_android webview_flutter_platform_interface webview_flutter_wkwebview]

dev dependencies:
- build_runner 2.1.5 [args async analyzer build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- chopper_generator 4.0.3 [analyzer build built_collection chopper code_builder dart_style logging meta source_gen]
- flutter_launcher_icons 0.9.2 [args image path yaml]
- flutter_lints 1.0.4 [lints]
- flutter_native_splash 1.3.2 [args image meta path xml yaml universal_io]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- json_serializable 6.1.1 [analyzer async build build_config collection json_annotation meta path pub_semver pubspec_parse source_gen source_helper]
- pigeon 1.0.12 [analyzer args collection meta path]
```

</details>

---

**Comment by darshankawar** [comment]
@CarlosVallsCiscar 
Is this coming from production or from debug mode ?
Is it possible for you to provide a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that we can use to verify this behavior ?

**Comment by CarlosVallsCiscar** [comment]
> @CarlosVallsCiscar
> Is this coming from production or from debug mode ?
> Is it possible for you to provide a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that we can use to verify this behavior ?

It is from debug mode.

I create a minimal project with the same bug.
[Project](https://github.com/CarlosVallsCiscar/firebase_perf_crash)

**Comment by russellwheatley** [comment]
Hi @CarlosVallsCiscar, I would double check these [values] (https://github.com/CarlosVallsCiscar/firebase_perf_crash/blob/master/lib/custom_client.dart#L20-L22) aren't `null` or invalid before setting.

**Comment by CarlosVallsCiscar** [comment]
> Hi @CarlosVallsCiscar, I would double check these [values] (https://github.com/CarlosVallsCiscar/firebase_perf_crash/blob/master/lib/custom_client.dart#L20-L22) aren't `null` or invalid before setting.

I have checked those values before setting and now it no longer crash.
`responseContentType` field was the one that produced the error, but is it defined as a `string?` and I think it shouldn't crash if I pass it a null value. I close the issue.

**Comment by CarlosVallsCiscar** [comment]
Sorry to reopen but I'm still having issues with HttpMetric.
I have been reviewing the code in Objective-C and in this class [FLTHttpMetric.m](https://github.com/FirebaseExtended/flutterfire/blob/051ff77b7e95c376dc2c5014877dd0a5a7856de8/packages/firebase_performance/firebase_performance/ios/Classes/FLTHttpMetric.m#L48-L59) the `null` is checked with` != nil`, but if i put a breakpoint, if I don't put for example `responsePayloadSize` what it receives from the arguments is an NSNull. I have not worked with Objective-C but I have searched a bit I have understood that `NSNull != nil` because `NSNull` is an instantiated object so this check is always `true` if I don't pass those optional values.

**Comment by chloe722** [comment]
I'm experiencing the same issue when using `HTTpMetric`. ios crash after calling `metric.stop()`
I use 0.8.0+2

**Comment by Salakar** [comment]
This was fixed in #7593 - pending release

**Comment by Salakar** [comment]
Released in `firebase_performance` `v0.8.0+3` - thanks
